includes/admin/class-wc-payments-admin-sections-overwrite.php

```diff
@@ -135,7 +135,7 @@ public function remove_overwrite_payments_tab_url_filter() {
 	 */
 	public function overwrite_payments_tab_url( $url, $path ): string {
 		if ( 'admin.php?page=wc-settings&tab=checkout' === $path ) {
-			return add_query_arg( [ 'section' => 'woocommerce_payments' ], $url );
+			return add_query_arg( [ 'section' => 'woocommerce_payments' ], $url ); // nosemgrep: audit.php.wp.security.xss.query-arg -- Admin area URL is passed in.
```

Comment:
Can we check if `$url` passed is being created dynamically by any external input?

Reply:
Its [from the admin_url](https://github.com/Automattic/woocommerce-payments/blob/a8503aa92ce68a97513feb49e0440b1feef022b4/includes/admin/class-wc-payments-admin-sections-overwrite.php#L114) filter, which passes the complete admin area URL including scheme and path according to [this documentation](https://developer.wordpress.org/reference/hooks/admin_url/#parameters)

Reply:
It internally uses [get_admin_url](https://developer.wordpress.org/reference/functions/get_admin_url/) so looks safe to me